pkg/handlers/wsclient.go

```diff
@@ -0,0 +1,158 @@
+package handlers
+
+import (
+	"encoding/json"
+	"github.com/gorilla/websocket"
+	"log"
+	"log/slog"
+	"strconv"
+	"time"
+)
+
+type WsClientList map[*WsClient]bool
+
+type WsClient struct {
+	// the websocket connection
+	connection *websocket.Conn
+
+	// manager is the manager used to manage the client
+	manager *WsServer
+	// egress is used to avoid concurrent writes on the WebSocket
+	egress chan Event
+	// tenantID is used to know what room user is in
+	tenantID string
+}
+
+var (
+	// pongWait is how long we will await a pong response from client
+	pongWait = 10 * time.Second
+	// pingInterval has to be less than pongWait, We cant multiply by 0.9 to get 90% of time
+	// Because that can make decimals, so instead *9 / 10 to get 90%
+	// The reason why it has to be less than PingRequency is becuase otherwise it will send a new Ping before getting response
+	pingInterval = (pongWait * 9) / 10
+)
+
+// NewClient is used to initialize a new Client with all required values initialized
```

Comment:
GoLang Doc Audit: comment on exported function NewWsClient should be of the form "NewWsClient ..." (ST1020)

```suggestion
// NewWsClient is used to initialize a new Client with all required values initialized
```